test/e2e/framework/addon/vault/proxy.go

```diff
@@ -55,7 +55,7 @@ func newProxy(ns, podName, kubectl string, vaultCA []byte) *proxy {
 }
 
 func (p *proxy) init() (*vault.Client, error) {
-	listenPort, err := p.freePort()
+	listenPort, err := testnet.FreePort()
```

Comment:
:+1: this works for now, but I'd really like for us to move away from requiring this function altogether in future if at all possible (i.e. by binding to `:0` and then inspecting the port number on the listener).

Perhaps a `NewRandomListener` function or something that we re-use, or just call `NewListener` here and retrieve the port.

Something to look into later 😄

Reply:
Ah, it looks like this function isn't actually used in the new code? Can we revert the change to create the testnet package if so? 😄